.gitignore

```diff
@@ -1,4 +1,388 @@
-bin/
```

Comment:
Should we be deleting this? Is it included in the below?
Why add all this below? Can we just add small bits as we go, keep it simple and add complexity progressively?